theano/gpuarray/dnn.py

```diff
@@ -2833,6 +2833,295 @@ def local_abstractconv3d_cudnn_graph(op, context_name, inputs, outputs):
+    """
+    __props__ = ('dtype',)
+    _cop_num_inputs = 6
+    _cop_num_outputs = 2
+    _f16_ok = True
+    default_output = 0
+
+    def __init__(self, dtype):
+        DnnBase.__init__(self, ["c_code/dnn_sptf.c"], "APPLY_SPECIFIC(dnn_sptf)")
+        self.dtype = dtype
+
+    def make_node(self, img, theta, output, desc, alpha=None, beta=None):
+        context_name = infer_context_name(desc)
+
+        img = gpu_contiguous(as_gpuarray_variable(img, context_name))
+        if img.type.ndim != 4:
+            raise TypeError('img must be a 4D tensor')
+        elif img.dtype not in ('float16', 'float32', 'float64'):
+            raise TypeError('img type must be floating-point')
+
+        theta = gpu_contiguous(as_gpuarray_variable(theta, context_name))
+        assert theta.dtype in ('float16', 'float32', 'float64')
+
+        # Setup grid dimensions using input from descriptor
+        grid_dims = as_tensor_variable(desc.owner.inputs[0])
+
+        output = gpu_contiguous(as_gpuarray_variable(output, context_name))
+        if output.type.ndim != 4:
+            raise TypeError('output must be a 4D tensor')
+
+        if (not isinstance(desc.type, CDataType) or
+                desc.type.ctype != 'cudnnSpatialTransformerDescriptor_t'):
+            raise ValueError('desc must be cudnnSpatialTransformerDescriptor_t')
+
+        alpha = ensure_dt(alpha, _one, 'alpha', img.dtype)
+        beta = ensure_dt(beta, _zero, 'beta', img.dtype)
+
+        grid = GpuArrayType(dtype=self.dtype,
+                            broadcastable=img.type.ndim * (False,),
+                            context_name=context_name)()
+
+        inputs = [img, theta, grid_dims, desc, alpha, beta]
+        outputs = [output.type(), grid]
+        return Apply(self, inputs, outputs)
+
+    def L_op(self, inputs, outputs, grads):
+        img, theta, output, desc, alpha, beta = inputs
+        _, grid = outputs
+        dy = grads[0]
+
+        dimg, dgrid = GpuDnnTransformerGradI(self.dtype)(img, theta, grid, dy,
+                                                         desc, alpha, beta)
+        dtheta = GpuDnnTransformerGradT(self.dtype)(dgrid, desc)
+        dalpha = theano.gradient.grad_not_implemented(self, 4, alpha)
+        dbeta = theano.gradient.grad_not_implemented(self, 5, beta)
+
+        return [dimg, dtheta, dy, DisconnectedType()(), dalpha, dbeta]
+
+    def connection_pattern(self, node):
+        # not connected to desc
+        return [[1, 1], [1, 1], [1, 1], [0, 0], [1, 1], [1, 1]]
+
+
+class GpuDnnTransformerGradI(DnnBase):
+    """
+    Gradients of inputs of the spatial transformer
+    """
+    __props__ = ('dtype',)
+    _cop_num_inputs = 8
+    _cop_num_outputs = 2
+    _f16_ok = True
+
+    def __init__(self, dtype=theano.config.floatX):
+        DnnBase.__init__(self, ["c_code/dnn_sptf_gi.c"], "APPLY_SPECIFIC(dnn_sptf_gi)")
+        self.dtype = dtype
+
+    def make_node(self, img, theta, grid, dy, desc, alpha, beta):
```

Comment:
Whenever alpha and beta are used, there must be an `output` in inputs list. Recall how alpha and beta are used:
```python
final_output = alpha * computed_output + beta * given_output
```
where `given_output` is the output given in `make_node()`.

**NB**: As it seems too complicated, I ultimately suggest you remove alpha and beta support. I suggested to support it to facilitate any future evolution of code, but in fact they are not currently used, event for cuDNN convolutions ops (at least in public interface). Sorry for that new request for interface change, but I think it will make your work easier.

So, remove alpha and beta support for transformer op and transformer grad ops. Just use `alpha=1` and `beta=0` in C code.